### PR TITLE
Move 4 functions from salt.utils to salt.utils.versions

### PR DESCRIPTION
**Pull request by terminalmage**
This moves the following functions from `salt.utils` to `salt.utils.versions`:

- `warn_until`
- `kwargs_warn_until`
- `compare_versions` (as `salt.utils.versions.compare`)
- `version_cmp`

### CODE REVIEW

**Comment by JensRantil** [comment]
I'll pass on this review, thanks.

**Comment by rallytime** [comment]
@terminalmage Looks like the tests aren't very happy here I am afraid. Can you take a look?

**Comment by terminalmage** [comment]
@rallytime Already in the middle of doing so ;)

**Comment by terminalmage** [comment]
Go Go Jenkins!

**Comment by terminalmage** [comment]
@rallytime there are still some issues here, but I'm going to pause on this PR and come back to it once I've sorted out the jenkins failures in 2016.11 and 2017.7.

File: salt/states/pkg.py [hunk 1]
```diff
@@ -80,10 +80,11 @@
 import os
 import re
 
-# Import salt libs
-import salt.utils
+# Import Salt libs
+import salt.utils  # Can be removed once gen_state_tag is moved
 import salt.utils.pkg
 import salt.utils.platform
+import salt.utils.versions
```
Comment (rallytime) [inline salt/states/pkg.py]: Where is this import change used? I don't see any `salt.utils.versions.x` functions anywhere. I do see `from salt.utils.versions import LooseVersion` below the windows gated section, but that's all. Am I missing something?

**Review by rallytime — COMMENTED** [review]
@terminalmage I only had one small comment/question.

File: salt/states/pkg.py (hunk 1 above)
Comment (terminalmage) [inline salt/states/pkg.py]: I missed making an update to one of the refs to `salt.utils.compare_versions`. Just pushed a commit to address this. Good catch!

Comment (rallytime) [inline salt/states/pkg.py]: 👍

**Review by rallytime — APPROVED** [review]

**Pull request merged**